File: 프로그래머스/lv0/120840. 구슬을 나누는 경우의 수/구슬을 나누는 경우의 수.swift
import Foundation




func solution(_ balls:Int, _ share:Int) -> Int64 {
    var min = balls - share < share ? balls - share : share

    var answer: Int64 = 1
    if min == 0 { return 1 }
    for i in 1...min {
        answer *= Int64(balls-min+i)
        answer /= Int64(i)
    }
    return answer
}

File: 프로그래머스/lv0/120840. 구슬을 나누는 경우의 수/구슬을 나누는 경우의 수.js
function solution(balls, share) {
    var answer = factorial(balls) / (factorial(balls-share) * factorial(share));
    return answer;
}
function factorial(num) {
    let ret = BigInt(1);
    for(let i=2; i<=num ; i++){
        ret *= BigInt(i);
    }
    return ret;
}